### PR TITLE
rapidcheck: fix gmock option and convert to v2

### LINKED ISSUE #14854: [package] rapidcheck/cci.20220514: cannot enable enable_gmock option
### Description

If I enable the `enable_gmock` option in `rapidcheck` the verification step always fails despite `gtest:build_gmock` is already enabled.

### Package and Environment Details

* Package Name/Version: **rapidcheck/cci.20220514**
* Conan version: **conan 1.55**
* Python version: **Python 3.10.8**

### Conan profile

```ini
[settings]
os=Macos
os_build=Macos
arch=x86_64
arch_build=x86_64
compiler=apple-clang
compiler.version=14
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce

```ini
[requires]
gtest/1.12.1
rapidcheck/cci.20220514

[options]
rapidcheck:enable_gtest=True
rapidcheck:enable_gmock=True
gtest:build_gmock=True
```

Results in
```
ERROR: rapidcheck/cci.20220514: Error in validate() method, line 74
	if self.options.enable_gmock and not self.deps_cpp_info["gtest"].build_gmock:
	KeyError: 'gtest'
```

### Logs

<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

WARN: rapidcheck/cci.20220514: requirement gtest/1.11.0 overridden by your conanfile to gtest/1.12.1 
ERROR: rapidcheck/cci.20220514: Error in validate() method, line 74
	if self.options.enable_gmock and not self.deps_cpp_info["gtest"].build_gmock:
	KeyError: 'gtest'
```

</details>

### PR DESCRIPTION
**Pull request by gjasny**
Specify library name and version:  **rapidcheck/cci.20220514**

This fixes #14854

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gjasny** [comment]
@memsharded @SpaceIm please review

**Comment by gjasny** [comment]
Please?

**Comment by SpaceIm** [comment]
Fix errors first please, v2 migration linter is red.

File: recipes/rapidcheck/all/conanfile.py [hunk 1]
```diff
@@ -33,32 +37,23 @@ class RapidcheckConan(ConanFile):
         "enable_gtest": False,
     }
```
Comment (RubenRBS) [inline recipes/rapidcheck/all/conanfile.py:39]: Would be great to have `package_type` here https://docs.conan.io/2/reference/conanfile/attributes.html?highlight=package_type#package-type

File: recipes/rapidcheck/all/conanfile.py [hunk 2]
```diff
@@ -33,32 +37,23 @@ class RapidcheckConan(ConanFile):
...
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.enable_catch:
```
Comment (RubenRBS) [inline recipes/rapidcheck/all/conanfile.py:59]: Could you try to bump the versions of these dependencies? `gtest` looks like a clean bump, but catch might be a bit trickier, let me know if you need any help with that!

File: recipes/rapidcheck/all/conanfile.py [hunk 3]
```diff
@@ -114,7 +110,7 @@ def package(self):
         )
 
     @staticmethod
```
Comment (RubenRBS) [inline recipes/rapidcheck/all/conanfile.py:112]: I think this might no longer need to be a `staticmethod` if you're passing self as the first argument :)

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot for your patience while we processed the PR backlog, sorry about the delay. The changes look great and I just had some minor sugestions before merging :)

**Comment by RubenRBS** [comment]
@NAThompson sorry about that, the Conan 2 launch has given us lots of work and we're still processing the backlog it generated. Thanks a lot for your patience :)

File: recipes/rapidcheck/all/conanfile.py (hunk 2 above)
Comment (gjasny) [reply recipes/rapidcheck/all/conanfile.py:59]: I bumped catch2 to latest 2.x version and gtest to latest c++11 compatible one.

File: recipes/rapidcheck/all/conanfile.py [hunk 4]
```diff
@@ -113,8 +109,7 @@ def package(self):
             }
         )
 
-    @staticmethod
-    def _create_cmake_module_alias_targets(module_file, targets):
+    def _create_cmake_module_alias_targets(self, conanfile, module_file, targets):
```
Comment (RubenRBS) [inline recipes/rapidcheck/all/conanfile.py:112]: self and conanfile are now the same :)

Comment (gjasny) [reply recipes/rapidcheck/all/conanfile.py]: Good catch. Fixed.

**Review by RubenRBS — APPROVED** [review]
Nice! Thank you so much again for your work on this, it's truly appreciated :)

File: recipes/rapidcheck/all/conanfile.py [hunk 5]
```diff
@@ -33,78 +37,70 @@ class RapidcheckConan(ConanFile):
...
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    def _min_cppstd(self):
+        return 11
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
```
Comment (SpaceIm) [inline recipes/rapidcheck/all/conanfile.py:51]: since this file is not required anymore, it should be removed

**Comment by SpaceIm** [comment]
Do not force push please.

**Review by RubenRBS — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**